luasteam.cxx

```diff
@@ -3,109 +3,210 @@
 #include <string>
 
 #ifdef _WIN32
-	#define EXTERN extern "C" __declspec( dllexport )
+#    define EXTERN extern "C" __declspec(dllexport)
 #else
-	#define EXTERN extern "C"
+#    define EXTERN extern "C"
 #endif
 
 extern "C" {
-#include <lua.h>
 #include <lauxlib.h>
+#include <lua.h>
 }
 
-//GENERAL FUNCTIONS
-EXTERN int luasteam_init(lua_State *L){
-    lua_settop(L,0); //Reset lua stack
-    lua_pushboolean(L,SteamAPI_Init());
-    return 1;
-}
+namespace {
 
-EXTERN int luasteam_shutdown(lua_State *L){
-    SteamAPI_Shutdown();
-    return 0;
-}
+class SteamFriendsListener;
+
+lua_State *L = nullptr;
```

Comment:
A global lua_State is something to avoid. If you need to this, at least give it another name so uses can be clearly identified.

Reply:
I needed this to be able to access the state from inside the callback. Will rename it.